docs/e2e-testing.md

```diff
@@ -0,0 +1,208 @@
+
+The process of creating an E2E test involves several steps: identification, development, testing, and maintenance. Initially, it is important to identify the right aspects of the product to test. The next step is to develop the tests, adhering to the quality guidelines outlined in this document. This includes creating clear test cases, using appropriate assertions, and structuring the tests for easy maintenance. Finally, it's crucial to ensure that tests are robust and reliable. This means the test should consistently produce the same results and be resilient to minor system changes. By following these steps, effective E2E tests can be created that contribute to delivering a high-quality product.
+
+This document serves as a guideline to approach writing E2E tests so that they provide fast feedback, they become less fragile over time, they are painless to debug, and they are scalable and easy to maintain. The remainder of this document will attempt to explain the suggested approach with relevant examples to help better understand the future direction for tests.
+
+These guidelines aren't meant to be a strict set of rules, they should be thought of as an approach to writing robust end-to-end tests. Use them where they make sense and adapt them as necessary to complement your tests.
+
+We have implemented two separate approaches for structuring tests on extensions and mobile. This document serves as a valuable reference point for establishing a common ground when writing end-to-end (E2E) tests across both clients. To gain a more comprehensive understanding of the specific guidelines for test writing on each platform, you can find detailed information here:
+
+- [Extension](e2e/extension-e2e-guidelines.md)
+- [Mobile](e2e/mobile-e2e-guidelines.md)
+
+Below, we present our standardized approach to writing tests that applies universally to both clients
+
+## Test names
+
+The test name should communicate the purpose and behaviour of the test. A clear test name serves as a concise summary of what the test aims to verify, making it easier for everyone to understand and maintain the tests. Clear test names improve the readability of the test and help when it comes to debugging a failed test. You should be able to figure out the purpose of the test without going through the complete implementation. If the test is failing, it should be easy to figure out which functionality is broken from the test name.
+
+### Guidelines
+
+✅ Recommended readable test names
+
+```javascript
+- adds Bob to the address book
+- send 1 TST to Bob
+```
+
+⚠️ Test names remains approachable, recommend to avoid using the prefix meaningless 'should'
+
+```javascript
+- should add Bob to the address book
+- should send 1 TST to Bob
+```
+
+❌ Test name should be completely avoided: meaningless `should` prefix, the `and` decreases the readability of the test,making it harder to understand what the test is doing as well as diagnose and fix issues.
```

Comment:
'Test names' instead of singular 'test name' because there are 2 tests.
'...of the test,making it harder..' - missing space after a comma.
'The use of a 'should' prefix and the word 'and' can decrease..' may be?

Reply:
Great point, fixed!